### DISCUSSION TITLE
Populating OneToOne

### DISCUSSION THREAD

**Discussion by Arcus16** [mikro-orm/mikro-orm#3313] [Q&A]
Hi, I have the following entities:

```javascript
export default class Resource extends BaseEntity<Resource, 'id'> {
    @PrimaryKey()
    id!: number;
    
     @OneToOne(() => LinkShare, linkShare => linkShare.resource, { nullable: true })
    linkShare!: LinkShare | null;
}
```

```javascript
export default class LinkShare extends BaseEntity<LinkShare, 'id'> {
    @PrimaryKey()
    id!: number;

    @OneToOne(() => Resource, resource => resource.linkShare, { owner: true, onDelete: 'cascade' })
    resource!: Resource;
}
```

When I call this:
```javascript
this.resourceRepository.findOne({ id }, { populate: ['linkShare'] })
```
and then somewhere else in code I call this:
```javascript
this.em.populate(resource, ['linkShare'])
```
Then two things can happen:
1. The LinkShare exists, and is already in the identity map eg. no additional query.
2. The LinkShare doesn't exist, a query will be made.

I don't understand why we need another query, when LinkShare is null?

**Comment by B4nan** [comment]
> I don't understand why we need another query, when LinkShare is null?

Because you dont know whether its null or not just from looking at the inverse side. We join the owner automatically to get its PK when you load the entity, to be able to populate things lazily as well as to know if its set or not, but still that relation could be set up after the entity was loaded. 

But thinking about this, maybe we could skip the query as it would be consistent with how other types of relation work (they also dont check for future updates and just use what's in the identity map). So as long as the `autoJoinOneToOneOwner` is true (default), we could skip it.
